### ISSUE TITLE
Reverse geocoding blocked by CSP for geopoint questions

### ISSUE THREAD

**Issue by derickl** [medic/cht-core#6184]
Related to #6183 

Geopoint questions, when rendered, present a textbox to allow users to reverse-geocode addresses of interest. The current setup is set to use google maps for this service but is currently blocked by CSP as per the error below:

```
inbox.js:3 Refused to connect to 'https://maps.googleapis.com/maps/api/geocode/json?address=kiserian&sensor=true' because it violates the following Content Security Policy directive: "connect-src 'self' https://staging.dev.medicmobile.org/_couch/builds/".
```
<img width="635" alt="Screenshot 2019-12-13 at 16 25 59" src="https://user-images.githubusercontent.com/83162/70803677-adcede80-1dc5-11ea-86a1-ebe2f5616fa1.png">

**Comment by garethbowen** [comment]
@derickl I've gotten around the initial error but now I'm getting:

```json
{
   "error_message" : "You must use an API key to authenticate each request to Google Maps Platform APIs. For additional information, please refer to http://g.co/dev/maps-no-account",
   "results" : [],
   "status" : "REQUEST_DENIED"
}
```

Is this feature necessary for a project you're working on, or just an error you noticed?

**Comment by derickl** [comment]
@garethbowen apologies for the delayed response. 
Some projects need a way to manually select a geopoint and I guess that the map view definitely
complement that.

**Comment by garethbowen** [comment]
I've created a new issue for the API access bug so we can move forward with the CSP fix: #6303

**Comment by dianabarsan** [comment]
Ready for AT on `6183-fix-csp-for-geopoint-widget`

**Comment by ngaruko** [comment]
LGTM
**master**
![image](https://user-images.githubusercontent.com/6979995/79954316-480c0500-84d1-11ea-8bfb-78d46a58698a.png)

 **branch**
![image](https://user-images.githubusercontent.com/6979995/79975561-481afd80-84ef-11ea-9833-efc37458c91d.png)

--- > How is the search functionality supposed to be working @derickl ? 
![image](https://user-images.githubusercontent.com/6979995/79975930-dd1df680-84ef-11ea-90cf-f475dc18498e.png)

**Comment by garethbowen** [comment]
@ngaruko I think the search functionality is covered by this issue: #6303

**Comment by binokaryg** [comment]
Is it also supposed to work with Android app?
Here is my observation in 3.9.0-beta.2 using `geopoint` type and changing the `appearance` column:


appearance | Mobile browser | Android app
-- | -- | --
<blank> | No map section | No map section
placement-map | No map section | No map section
maps | Works after clicking Map button | Map button shows but map image is blank

**Comment by garethbowen** [comment]
> Is it also supposed to work with Android app?

Yes it should work. I tested it with no appearance. The widget relies on Google Maps so it's possible it's rate limiting or blocking you, and it will not work offline.

Can you check the browser logs and see if there are any errors reported?

**Comment by binokaryg** [comment]
The geopoint field works, as it gets the coordinates, but no map is shown when I am using the Android app with no appearance. I couldn't find any errors in the logs.

![image](https://user-images.githubusercontent.com/20041732/84677287-3e8d9f00-af4e-11ea-89f1-5bd8433cb400.png)

**Comment by garethbowen** [comment]
@binokaryg Thanks! Can you confirm you were online when loading the form?

To help us assess the priority, is having the map required for a project, or are you just trying it out?

Do you want to interact with the map so you can for example manually enter a location, or is it useful to see a static image of where the phone thinks it is?

**Comment by binokaryg** [comment]
Yes, I was online when loading the form.
I guess the priority may not be that high. The requirement does not include manually entering a location. However, it might be helpful for the CHWs to confirm that the coordinates match the location they are trying to record.

**Comment by mrjones-plip** [comment]
In relation to the map and search field simply not appearing at all, I have maybe some helpful input. I was doing some testing for [some changes](https://github.com/medic/config-moh-mali/issues/91) in CMoHM app. As expected, no map at all is shown with default path of cht-core 3.9.1 with and medic-android 0.4.39, I found the same to be true on Android 10 with Chrome 85.0.4183.127. 

Because the map didn't show in new Chrome on mobile, this got me thinking about reproducing on the desktop.  Indeed, if you are in the desktop (Ubuntu 18 for me) and test in either FF 81.0  or Chrome 85.0.4183.121, the map shows as expected.  I was confused at first because switching to the "Pixel 2" profile in developer tools did not make the map disappear as I expected it to.  Instead fully reloading the page (shift + reload) caused the map to disappear.  Inspecting the DOM, I can see the map elements are fully not there in responsive mode.  I didn't chase it all down, but this means there's some sort of detection going on at initial page load to add these elements to the DOM or not.  This is _not_ media sensitive CSS hiding or showing based on screen resolution.

NB - I never saw the map box render but show empty like initially reported.  When it was present, it always worked.
